src/Workspaces/Core/Portable/Utilities/AsyncLazy`1.cs

```diff
@@ -520,7 +520,10 @@ private void OnAsynchronousRequestCancelled(object state)
             }
         }
 
-        private sealed class Request
+        /// <remarks>
+        /// This inherits from <see cref="TaskCompletionSource{TResult}"/> to avoid allocating two objects when we can just use one.
+        /// </remarks>
```

Comment:
i don't hate this.  and less allocations are better.  But, for some reason, i prefer the old containment pattern vs inheritance.